src/main/java/org/wso2/carbon/identity/conditional/auth/typingdna/SaveUserInTypingDNAFunctionImpl.java

```diff
@@ -0,0 +1,108 @@
+
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import javax.net.ssl.HttpsURLConnection;
+
+/**
+ * Custom adaptive function implementation for Save
+ * users typing pattern in TypingDNA.
+ */
+public class SaveUserInTypingDNAFunctionImpl implements SaveUserInTypingDNAFunction {
+
+    private static final Log log = LogFactory.getLog(VerifyUserWithTypingDNAFunctionImpl.class);
+
+    @Override
+    public void saveUserInTypingDNA(JsAuthenticationContext context) throws Exception {
+
+        JsAuthenticatedUser user = utils.getUser(context);
+        String username = user.getWrapped().getUserName();
+        String tenantDomain = user.getWrapped().getTenantDomain();
+        String typingPattern = utils.getTypingPattern(context);
+
+        // Getting connector configurations.
+        String APIKey = CommonUtils.getConnectorConfig(TypingDNAConfigImpl.USERNAME, tenantDomain);
+        String APISecret = CommonUtils.getConnectorConfig(TypingDNAConfigImpl.CREDENTIAL, tenantDomain);
+        String advanced = CommonUtils.getConnectorConfig(TypingDNAConfigImpl.ADVANCE_MODE_ENABLED, tenantDomain);
+        String region = CommonUtils.getConnectorConfig(TypingDNAConfigImpl.REGION, tenantDomain);
+
+        String userID = getUserID(username, tenantDomain);
+
+        if (!(typingPattern == null) && !typingPattern.equals(Constants.NULL)) {
+            String baseurl = buildURL(region, userID);
+            String data = "tp=" + URLEncoder.encode(typingPattern, "UTF-8");
+            String Authorization = Base64.getEncoder().encodeToString((APIKey + ":" + APISecret).getBytes(StandardCharsets.UTF_8));
+
+            // Setting up URL connection.
+            URL url = new URL(baseurl);
+            HttpsURLConnection connection = (HttpsURLConnection) url.openConnection();
+            connection.setRequestMethod("POST");
+            connection.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");
+            connection.setRequestProperty("Authorization", "Basic " + Authorization);
+
+            connection.setUseCaches(false);
+            connection.setDoOutput(true);
+
+            DataOutputStream wr = new DataOutputStream(connection.getOutputStream());
+            wr.writeBytes(data);
+            wr.close();
+
+            InputStream is = connection.getInputStream();
+            BufferedReader rd = new BufferedReader(new InputStreamReader(is));
+            StringBuilder res = new StringBuilder();
+            String line;
+            while ((line = rd.readLine()) != null) {
+                res.append(line);
+                res.append('\r');
+            }
+            rd.close();
+
+            // Response from TypingDNA
+            log.debug(res.toString());
```

Comment:
You need to check whether the debug mode is enabled. Also, format the log in a proper manner.